lib/ansible/plugins/action/net_base.py

```diff
@@ -127,6 +136,57 @@ def run(self, tmp=None, task_vars=None):
 
         return result
 
+    def _nxapi_implementation(self, play_context):
```

Comment:
To avoid code duplication I think this logic can be implemented as a function in individual action plugins. Thoughts?